modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BInteger.java

```diff
@@ -24,20 +24,20 @@
  */
 public final class BInteger extends BValueType {
 
-    private int value;
+    private long value;
 
-    public BInteger(int value) {
+    public BInteger(long value) {
         this.value = value;
     }
 
     @Override
     public int intValue() {
-        return this.value;
+        return (int) this.value;
```

Comment:
Same suggestion as above. return long value from this method and remove the longValue() method entirely.